sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/NewGuidShouldNotBeUsed.cs

```diff
@@ -0,0 +1,59 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class NewGuidShouldNotBeUsed : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S4581";
+        private const string MessageFormat = "Please review this use of 'new Guid()'.";
```

Comment:
I would rather suggest what we think is wrong `Use "Guid.NewGuid() or "Guid.Empty" instead of "new Guid()"`

Reply:
@duncanp-sonar @valhristov Would it be ok: `Use 'Guid.NewGuid()', 'Guid.Empty' or add arguments to this Guid instantiation.`?